hip-0000-template.md

```diff
@@ -1,59 +1,276 @@
+
+Wallet and token explorer implementations interrogate HTS tokens using this standard to display additional metadata for tokens.
 
 ## Reference Implementation
 
-The reference implementation must be complete before any HIP is given the status of “Final”. The final implementation must include test code and documentation.
+This section provides examples of a metadata schema adhering to this specification.
+
+### Example Schema: Required and Recommended fields ONLY
+
+```json
+{
+    "name": "Redhill Farm Carbon Credit",
+    "creator": "Redhill Farm",
+    "format": "guardian-dea",
+    "description": "This is a carbon credit issued by Redhill Farm",
+    "image": "ipfs://bafybeiapq5ewrpxrsche54laxvwwvth2ye3df4wbswwe3z2mk7rgzlz7za",
+    "type": "image/png",
+    "properties": {
+        "guardian_proof": {
+            "type": "Ed25519Signature2018",
+            "created": "2023-10-04T11:18:59Z",
+            "verificationMethod": "did:hedera:testnet:Ci3QBHbe3Fa6gKn8hm44PJn1uMkXEDbJkPQboYN7EKWV_0.0.3160930#did-root-key",
+            "proofPurpose": "assertionMethod",
+            "jws": "eyJhbGciOiJFZERTQSIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..c3Jmwg3Nis5puQLNh5zcMgkzmmsB-m10A9dOoGh_3tt8aaTIzMZSsxQJbiLmb795mw-sKfa9DqN7gy3kbgNDCA"
+        }
+    },
+    "attributes": [
+        {
+            "trait_type": "timestamp",
+            "value": 1710271999.591571
+        }
+    ]
+}
+```
+
+### Example Schema: Full-featured
 
+```json
+{
+    "name": "Redhill Farm Carbon Credit",
+    "creator": "Redhill Farm",
+    "format": "guardian-dea",
+    "description": "This is a carbon credit issued by Redhill Farm",
+    "image": "ipfs://bafybeiapq5ewrpxrsche54laxvwwvth2ye3df4wbswwe3z2mk7rgzlz7za",
+    "type": "image/png",
+    "attributes": [
+        {
+            "trait_type": "timestamp",
+            "value": "1710271999.591571003"
+        },
+        {
+            "trait_type": "asset_type",
+            "value": "CRU"
+        },
+        {
+            "trait_type": "sector",
+            "value": "Agriculture"
+        },
+        {
+            "trait_type": "project_type",
+            "value": "Agriculture"
+        },
+        {
+            "trait_type": "method",
+            "value": "removal"
+        },
+        {
+            "trait_type": "vintage",
+            "value": 2023
+        },
+        {
+            "trait_type": "geography",
+            "value": "UK"
+        },
+        {
+            "trait_type": "methodology_developer",
+            "value": "UNFCCC"
+        },
+        {
+            "trait_type": "methodology",
+            "value": "https://cdm.unfccc.int/methodologies/DB/99QRTE6N5QJEBOV2XP374B25SSIXBB"
+        },
+        {
+            "trait_type": "co-benefits",
+            "value": [
+                "air",
+                "water",
+                "soil",
+                "biodiversity",
+                "equity",
+                "carbon"
+            ]
```

Comment:
HIP 412 doesn't support arrays so we'll need to raise this as a new value type that can be used.

Reply:
Existing explorers would need to make changes to their filtering features to support arrays here. 

The alternative is to list them all at the root level which would allow them to act as filters without any change to existing NFT markets.

```
{
    "trait_type": "co-benefits-air",
    "display_type": "boolean",
    "value": true
},
{
    "trait_type": "co-benefits-water",
    "display_type": "boolean",
    "value": true
},
```

I do prefer the array solution originally proposed, just be aware that it can't be used as a filter out of the gate until markets update to support it.

Reply:
I like this.